pages/docs/tutorials/native/using-intellij-idea.md

```diff
@@ -1,50 +1,126 @@
 
-![Wizard Step Two]({{ url_for('tutorial_img', filename='native/using-intellij-idea/wizard-2.png')}})
+Your project opens. By default, the wizard creates the necessary `main.kt` file with the code that prints "Hello, Kotlin/Native!" to the standard output.
 
-On clicking **Next**, enter the path and name for the project.
+The `build.gradle.kts` file contains the project settings. You can change the settings using [Kotlin Multiplatform Gradle DSL reference](../../reference/mpp-dsl-reference.html).
 
-![Wizard Step Three]({{ url_for('tutorial_img', filename='native/using-intellij-idea/wizard-3.png')}})
+## Run the application
 
-This will complete the process and open the newly created project in the IDE. By default the wizard will create the necessary
-`Sample<TARGET>.kt` file and provide code for writing some string to the standard output. Note that `<TARGET>` varies based on the operating
-system the project was created on (Windows, Linux, macOS).
+Start the application by clicking **Run** next to the run configuration at the top of the screen.
 
-![Project]({{ url_for('tutorial_img', filename='native/using-intellij-idea/IDE-1.png')}})
+![Run the application]({{ url_for('tutorial_img', filename='native/using-intellij-idea/native-new-project-intellij-3.png') }})
 
-To run the project, simply invoke the [Run command in the IDE](https://www.jetbrains.com/help/idea/running-applications.html), by using the corresponding shortcuts or from the Run menu
+IntelliJ IDEA opens the **Run** tab and shows the output:
+![Application output]({{ url_for('tutorial_img', filename='native/using-intellij-idea/native-new-project-intellij-4.png') }})
 
-![Run]({{ url_for('tutorial_img', filename='native/using-intellij-idea/IDE-2.png')}})
+## Update the application
+
+### Count letters in your name
+
+1. Open the file `main.kt` in `src/<your_app_name>Main/kotlin`.
+
+   The `src` directory contains Kotlin source files and resources. The file `main.kt` includes sample code that prints "Hello, Kotlin/Native!" using the [`prinln()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/println.html) function.
+
+2. Change the code of `main()`:
+
+   * Use the [`readLine()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/read-line.html) function to read the input value to the `name` variable.
+   * Perform the null check of the input value with the [safe call operator `?.`](https://kotlinlang.org/docs/reference/null-safety.html#safe-calls).
+   * Use the [`replace()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/replace.html) function to eliminate the whitespaces in the name.
+   * Use the scope function [`let`](https://kotlinlang.org/docs/reference/scope-functions.html#let) to execute function within the object context. 
+   * Use a [string template](https://kotlinlang.org/docs/reference/basic-types.html#string-templates) to insert your name into the string by adding a dollar sign `$` and enclosing it in curly braces – `${it?.length}`.
```

Comment:
Again extra `?`. Variable is already not nullable. May be it's worth to mention that it is default name of lambda parameter. It may be unclear for beginners.

Reply:
Fixed